### DISCUSSION TITLE
style for BaseEdgeProps

### DISCUSSION THREAD

**Discussion by prksean** [bcakmakoglu/vue-flow#908] [Q&A]
(using "@vue-flow/core": "^1.18.0")

`export interface BaseEdgeProps extends EdgeLabelOptions {
  labelX?: number
  labelY?: number
  path: string
  label?: any
  markerStart?: string
  markerEnd?: string
  interactionWidth?: number
}`

I didn't look close enough to the source code but any reason why style is not added to BaseEdgeProps?
`style?: any` 
Trying to create custom edge with some style 
`<BaseEdge :path="path[0]" :style="style"/>`

But, getting such error (still functions but trying to avoid such vue-tsc error console)

> Argument of type '{ path: string; style: any; }' is not assignable to parameter of type 'BaseEdgeProps'. Object literal may only specify known properties, and 'style' does not exist in type 'BaseEdgeProps'.

By the way, thanks bunch for providing such a great component. I hope this great work continues!

**Comment by prksean** [comment]
I guess I can extend BaseEdgeProps and problem solved...

`<BaseEdge v-bind="customEdgeProps"/>`

`interface CustomEdgeProps extends BaseEdgeProps {
    style: any
}`

`const customEdgeProps = computed<CustomEdgeProps>( ... )`

**Comment by bcakmakoglu** [comment]
The reason why `style` is *not* a prop is because it's an attribute.
You can just pass it without it having to be explicitly defined as a prop on the BaseEdge component.

**Reply by bcakmakoglu** [reply]
Same goes for `class` btw.

**Reply by prksean** [reply]
Right, I can still pass it and still works. But I was trying to avoid the error that I mentioned. Anyways, thanks for your response!

**Reply by bcakmakoglu** [reply]
What version of vue-tsc are you using, as I'm not getting this error on `^1.2.0`
<img width="249" alt="image" src="https://user-images.githubusercontent.com/78412429/236167662-63ab0c4d-344e-4df3-8fc8-079a350b55cc.png">

**Reply by prksean** [reply]
Using 1.4.4

**Reply by bcakmakoglu** [reply]
I'll take a look, maybe this is a bug caused by vue-tsc, seeing as in older versions this issue does not appear :)
